source/pool/contracts/Pool.sol

```diff
@@ -116,34 +174,105 @@ contract Pool is Ownable {
    * @param token IERC20
    */
   function withdraw(Claim[] memory claims, IERC20 token) external {
-    withdrawProtected(claims, token, 0);
+    withdrawProtected(claims, token, 0, msg.sender, msg.sender);
+  }
+
+  /**
+   * @notice Withdraw tokens from the pool using claims and send to recipient
+   * @param claims Claim[]
+   * @param token IERC20
+   * @param recipient address
+   */
+  function withdrawWithRecipient(
+    Claim[] memory claims,
+    IERC20 token,
+    address recipient
+  ) external {
+    withdrawProtected(claims, token, 0, msg.sender, recipient);
+  }
+
+  /**
+   * @notice Withdraw tokens from the pool using claims and stake
+   * @param claims Claim[]
+   * @param token IERC20
+   */
+  function withdrawAndStake(Claim[] memory claims, IERC20 token) external {
+    require(token == IERC20(stakingToken), "INVALID_TOKEN");
+    uint256 amount = withdrawProtected(
+      claims,
+      token,
+      0,
+      msg.sender,
+      msg.sender
+    );
+    (bool success, ) = address(stakingContract).call(
+      abi.encodeWithSignature("stakeFor(address,uint256)", msg.sender, amount)
+    );
+    require(success, "ERROR_STAKING");
+  }
+
+  /**
+   * @notice Withdraw tokens from the pool using claims and stake for another account
+   * @param claims Claim[]
+   * @param token IERC20
+   * @param account address
+   */
+  function withdrawAndStakeFor(
+    Claim[] memory claims,
+    IERC20 token,
+    address account
+  ) external {
+    require(token == IERC20(stakingToken), "INVALID_TOKEN");
+    uint256 amount = withdrawProtected(
+      claims,
+      token,
+      0,
+      msg.sender,
+      msg.sender
+    );
+    (bool success, ) = address(stakingContract).call(
+      abi.encodeWithSignature("stakeFor(address,uint256)", account, amount)
+    );
+    require(success, "ERROR_STAKING");
   }
 
+  /**
+   * @notice Withdraw tokens from the pool using claims
+   
+   * @param claims Claim[]
+   * @param token IERC20
+   * @param minimumAmount uint256
+   * @param account address
+   * @param recipient address
+   */
   function withdrawProtected(
     Claim[] memory claims,
     IERC20 token,
-    uint256 minimumAmount
-  ) public {
+    uint256 minimumAmount,
+    address account,
```

Comment:
Two thoughts to discuss in more detail:
1. If Bob has Alice's claims, could he call this function directly with her as the `account` and himself as the `recipient` and take her reward?
2. If `account` is always passed as `msg.sender` would it be most clear to just use `msg.sender` in this function as in the previous setup?

Reply:
Yes to both of those points. I'll remove account and replace it with msg.sender in that withdrawProtected function.